### PR TITLE
Adding Doxygen description to some files in core

### PR DESCRIPTION
**Pull request by loumalouomega**

### CODE REVIEW

File: kratos/includes/serializer.h [hunk 1]
```diff
@@ -99,8 +99,11 @@ template <class TDataType> class Variable;
 ///@name Kratos Classes
 ///@{
 
-/// Short class definition.
-/** Detail class definition.
+/// The serialization consist of storing the state of an object into a storage format like data file or memory buffer and also retrieving the object from such a media.
```
Comment (rubenzorrilla) [inline kratos/includes/serializer.h]: typo

File: kratos/utilities/mortar_utilities.h [hunk 2]
```diff
@@ -45,7 +45,7 @@ namespace Kratos
 ///@name  Functions
 ///@{
     
-class MortarUtilities
+class KRATOS_API(KRATOS_CORE) MortarUtilities
```
Comment (roigcarlo) [inline kratos/utilities/mortar_utilities.h]: Remove KRATOS_API(KRATOS_CORE). This doesn't need to be exported and will crash on win.

Comment (loumalouomega) [reply kratos/utilities/mortar_utilities.h]: I though it was the opposite

**Comment by loumalouomega** [comment]
Done

**Review by pooyan-dadvand — APPROVED** [review]
:+1: 

Thanks!

**Comment by loumalouomega** [comment]
👍

**Pull request merged**